cmd/ingress/main.go

```diff
@@ -49,68 +51,73 @@ type envConfig struct {
 	BrokerURL    string `envconfig:"BROKER_URL" required:"true"`
 	ExchangeName string `envconfig:"EXCHANGE_NAME" required:"true"`
 
-	channel *amqp.Channel
+	connection *wabbitamqp.Conn
+	channel    *wabbitamqp.Channel
 }
 
 func main() {
 	ctx := signals.NewContext()
-
 	var env envConfig
 	if err := envconfig.Process("", &env); err != nil {
 		logging.FromContext(ctx).Fatal("Failed to process env var: ", err)
 	}
 
 	env.SetComponent(component)
+	var err error
 	logger := env.GetLogger()
 	ctx = logging.WithLogger(ctx, logger)
-	if err := env.SetupTracing(); err != nil {
-		logger.Errorw("Failed setting up trace publishing", zap.Error(err))
-	}
-	if err := env.SetupMetrics(ctx); err != nil {
-		logger.Errorw("Failed to create the metrics exporter", zap.Error(err))
-	}
 
-	conn, err := amqp.Dial(env.BrokerURL)
-	if err != nil {
-		logger.Fatalw("failed to connect to RabbitMQ", zap.Error(err))
+	if err = env.SetupTracing(); err != nil {
+		logger.Errorw("Failed setting up trace publishing", zap.Error(err))
 	}
-	defer conn.Close()
 
-	env.channel, err = conn.Channel()
-	if err != nil {
-		logger.Fatalw("failed to open a channel", zap.Error(err))
+	if err = env.SetupMetrics(ctx); err != nil {
+		logger.Errorw("Failed to create the metrics exporter", zap.Error(err))
 	}
 
+	retryChannel := make(chan bool)
+	env.CreateRabbitMQResources(retryChannel, logger)
 	// noWait is false
-	if err := env.channel.Confirm(false); err != nil {
-		logger.Fatalf("faild to switch connection channel to confirm mode: %s", err)
-	}
-	defer env.channel.Close()
+	if err = env.channel.Confirm(false); err != nil {
```

Comment:
if there's an error during `env.CreateRabbitMQResources`, it's quietly ignored and env.channel is unset. that would cause a nil pointer exception here